react-frontend/src/QASM/grid_utils.js

```diff
@@ -715,4 +726,49 @@ async function getImageStack(
         console.log(new_image_stack);
     }
     return new_image_stack;
+}
+
+/**
+ * Show or hide the center line for an image
+ * 
+ * @param {*} document document object
+ * @param {string} hover_image_id id of the current image
+ */
+export function toggleCenterLine(document, hover_image_id) {
+    if (hover_image_id === null) {
+        return;
+    }
+    
+    let center_line = document.getElementById(hover_image_id + "-center-line-overlay");
+    center_line.classList.toggle("hidden");
+}
+
+/**
+ * Show or hide all center lines for all images
+ * 
+ * @param {*} document 
+ * @param {*} component 
+ */
+export function toggleAllCenterLines(document, component) {
+    // Add a component property to keep track of the center line state
+    if (component.center_lines_hidden === undefined) {
+        component.center_lines_hidden = false;
+    }
+
+    // Toggle the property
+    component.center_lines_hidden = !component.center_lines_hidden;
+
+    if (component.center_lines_hidden) {
+        // Show all center lines
+        for (let image_name of component.image_names) {
+            let center_line = document.getElementById(image_name + "-center-line-overlay");
+            center_line.classList.remove("hidden");
+        }
+    } else {
+        // Hide all center lines
+        for (let image_name of component.image_names) {
+            let center_line = document.getElementById(image_name + "-center-line-overlay");
+            center_line.classList.add("hidden");
+        }
+    }
 }
```

Comment:
```suggestion
export function toggleAllCenterLines(document, component) {
    // Invert the center_lines_hidden state
    // If it was undefined it will be set to true
    component.center_lines_hidden = !Boolean(component.center_lines_hidden);
    const is_hidden = component.center_lines_hidden; // To shorten the variable name
    
    for (let image_name of component.image_names) {
        let center_line = document.getElementById(image_name + "-center-line-overlay");
        is_hidden ? center_line.classList.add("hidden") : center_line.classList.remove("hidden");
    }
}
```
Previously the `center_lines_hidden` property seems to be the opposite of the true state of the overlays. When `center_lines_hidden` is `True` the overlays stop being hidden, and vice versa. It doesn't actually matter for functionality but I figured it was worthwhile to fix. While I was here I figured I'd rewrite the code to be easier to read, but that is subjective.